packages/core/contracts/zodiac/OptimisticOracleModule.sol

```diff
@@ -0,0 +1,197 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.6;
+
+import "@gnosis.pm/zodiac/contracts/core/Module.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "../oracle/implementation/Constants.sol";
+import "../oracle/interfaces/FinderInterface.sol";
+import "../oracle/interfaces/SkinnyOptimisticOracleInterface.sol";
+import "../oracle/interfaces/OracleAncillaryInterface.sol";
+
+contract OptimisticOracleModule is Module {
+    event OptimisticOracleModuleDeployed(address indexed owner, address indexed avatar, address target);
+
+    event TransactionsProposed(uint256 indexed proposalId, address indexed proposer, uint256 indexed proposalTime);
+
+    event TransactionExecuted(uint256 indexed proposalId, uint256 indexed transactionIndex);
+
+    event ProposalDeleted(uint256 indexed proposalId);
+
+    FinderInterface public finder;
+    SkinnyOptimisticOracleInterface optimisticOracle;
+
+    IERC20 public collateral;
+    uint256 public bond;
```

Comment:
If `bond` can only be set by the DAO and the final fee changes to a value higher than bond, then could the DAO end up in a locked state where it cannot get an answer from the oracle and cannot update the required bond?

Reply:
Yeah great point. @pemulis i think we should save the bond per proposal and then before executing any proposal check if the bond has changed. 

If the bond has changed you can either cancel the proposal or ignore the change and just deal with that risk. But we definitely don't want to get locked up

Reply:
I think that would work. We could also change the parameter saved in the module to `additionalBond` so that the DAO can optionally require an additional bond above and beyond the final fee, and the combined bond is required to be approved and transferred by the user when making a proposal. Will tinker with this a bit.

Reply:
> If the user were to pass in bond as a parameter to proposeTransactions(), wouldn't that solve the issue?

Is that a conflict of interest? Should the proposer be able to pick their own bond?

> i think we should save the bond per proposal and then before executing any proposal check if the bond has changed.

I think the OO does this for us!

To the original question, I think this actually works as it's implemented now. The `bond` value passed to the OO (I think!) is _in addition to_ the final fee so the bond will never end up being lower than the final fee. The caller will be required to approve/send the bond + finalFee to the OO.

Reply:
From Matt's comment, this may work as-is! I'll double-check how this function interacts with the OO's bond handling logic and add some comments to the code in the module to make it clear that proposers have to post the bond *plus* the OO's final fee for the module's collateral type, which can be found in UMA's contracts and documentation.

Reply:
Here's the line where the optional bond is added to the final fee. Agree with Matt's assessment that this should work as-is and I'll add comments to the module contract that explain this.